server/src/test/java/org/elasticsearch/gateway/ReplicaShardAllocatorTests.java

```diff
@@ -369,6 +461,11 @@ public boolean getFetchDataCalledAndClean() {
         }
 
         public TestAllocator addData(DiscoveryNode node, String syncId, StoreFileMetaData... files) {
+            return addData(node, Collections.emptyList(), syncId, files);
```

Comment:
Should we add random retention lease info here, to check that it is unused?

Reply:
We can break some existing tests if we add some random leases for the primary shard.